### PR TITLE
feat: Prepare Permission core (ngrx) #6116 + unit tests #6115

### LINKED ISSUE #6116: Prepare Permission core (ngrx)

### PR DESCRIPTION
**Pull request by Pucek9**
Closes #6116 #6115

### CODE REVIEW

File: projects/core/src/organization/facade/permission.service.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+          permissionCode,
+          permission,
+        })
+      )
+    );
+  }
+
+  private withUserId(callback: (userId: string) => void): void {
+    if (this.authService) {
+      this.authService
```
Comment (marlass) [inline projects/core/src/organization/facade/permission.service.ts]: It always will be provided. Not need for if

File: projects/core/src/organization/model/search-config.ts [hunk 2]
```diff
@@ -4,4 +4,4 @@ export interface B2BSearchConfig {
   sort?: string;
 }
 
-export const LIST = 'list';
+export const ALL = 'all';
```
Comment (marlass) [inline projects/core/src/organization/model/search-config.ts]: Isn't it to generic?

File: projects/core/src/organization/store/actions/permission.action.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+export class LoadPermission extends EntityLoadAction {
+  readonly type = LOAD_PERMISSION;
+  constructor(public payload: { userId: string; permissionCode: string }) {
+    super(PERMISSION_ENTITIES, payload.permissionCode);
+  }
+}
+
+export class LoadPermissionFail extends EntityFailAction {
+  readonly type = LOAD_PERMISSION_FAIL;
+  constructor(permissionCode: string, public payload: any) {
```
Comment (marlass) [inline projects/core/src/organization/store/actions/permission.action.ts]: To make action more extendable it's better to have single parameter as an object. Then from payload you can extract whatever you need here.

File: projects/core/src/organization/store/actions/permission.action.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    super(PERMISSION_ENTITIES, permissionCode, payload);
+  }
+}
+
+export class LoadPermissionSuccess extends EntitySuccessAction {
+  readonly type = LOAD_PERMISSION_SUCCESS;
+
+  constructor(public payload: Permission[]) {
```
Comment (marlass) [inline projects/core/src/organization/store/actions/permission.action.ts]: Also here it's better to wrap it in object for easier extendability. Adding another property to array won't be easy.

File: projects/core/src/organization/model/search-config.ts (hunk 2 above)
Comment (Pucek9) [reply projects/core/src/organization/model/search-config.ts]: we wrap ids in serialized params (query), but what if api is not parameterized? That's my propose for object name for all available data

**Pull request merged**